### PR TITLE
Module-Oriented Project

### PR DESCRIPTION
**Pull request by jonathan016**
This pull request contains the project developed using modules. Previously, the project was developed using packages, but now the project is developed using modules.

### CODE REVIEW

File: .idea/.name [hunk 1]
```diff
@@ -0,0 +1 @@
+OASIS
```
Comment (ensist) [inline .idea/.name:1]: please delete all folder `.idea` from repository.
you can use `.gitignore` to ignore all folders or files that are not related to project

File: exception/src/main/java/com/example/exception/UserNotFoundException.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.example.exception;
```
Comment (ensist) [inline exception/src/main/java/com/example/exception/UserNotFoundException.java:1]: No need to create a new module for exception, just put it in service module

Comment (ensist) [inline exception/src/main/java/com/example/exception/UserNotFoundException.java:1]: please rename the package's name to be more meaningful, like com.oasis

File: model/src/main/java/com/example/model/EmployeeModel.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import lombok.Setter;
+import org.bson.BsonTimestamp;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.mapping.Document;
+
+import java.math.BigInteger;
+import java.util.Date;
+
+@Document(collection = "employees")
+@Getter @Setter
```
Comment (ensist) [inline model/src/main/java/com/example/model/EmployeeModel.java:14]: change @Getter and @Setter to @Data

File: model/src/main/java/com/example/model/EmployeeModel.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import lombok.Getter;
+import lombok.Setter;
+import org.bson.BsonTimestamp;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.mapping.Document;
+
+import java.math.BigInteger;
+import java.util.Date;
+
+@Document(collection = "employees")
```
Comment (ensist) [inline model/src/main/java/com/example/model/EmployeeModel.java:13]: create a constant for collection name

File: model/src/main/java/com/example/model/EmployeeModel.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+@Getter @Setter
+public class EmployeeModel {
+    @Id
+    @Setter(AccessLevel.PRIVATE)
+    private BigInteger _id;
+
+    private String employeeFullname;
```
Comment (ensist) [inline model/src/main/java/com/example/model/EmployeeModel.java:20]: no need prefix "employee", just rename it to fullname, dob, username, etc

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.example</groupId>
+    <artifactId>oasis</artifactId>
+    <version>1.0-SNAPSHOT</version>
+    <modules>
+        <module>exceptions</module>
```
Comment (ensist) [inline pom.xml:11]: remove exceptions module

File: pom.xml [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        <module>model</module>
+        <module>repository</module>
+        <module>service</module>
+        <module>webcontroller</module>
```
Comment (ensist) [inline pom.xml:15]: rename to web-controller

File: pom.xml [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+        <module>webmodel</module>
```
Comment (ensist) [inline pom.xml:16]: rename to web-model

File: repository/pom.xml [hunk 9]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.example</groupId>
+    <artifactId>repository</artifactId>
+    <version>1.0-SNAPSHOT</version>
```
Comment (ensist) [inline repository/pom.xml:9]: please change the version to 1.0.0-SNAPSHOT

File: repository/pom.xml [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+    <packaging>war</packaging>
+
+    <dependencies>
+        <!--Spring MongoDB Data Dependency-->
+        <dependency>
+            <groupId>org.springframework.data</groupId>
+            <artifactId>spring-data-mongodb</artifactId>
+            <version>2.1.0.RELEASE</version>
```
Comment (ensist) [inline repository/pom.xml:17]: dependency's version is only set in parent POM, in `<dependencyManagement>` section

File: repository/pom.xml [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+        </dependency>
+
+        <!--Model Dependency-->
+        <dependency>
+            <groupId>com.example</groupId>
+            <artifactId>model</artifactId>
+            <version>1.0-SNAPSHOT</version>
```
Comment (ensist) [inline repository/pom.xml:24]: dependency's version is only set in parent POM, in `<dependencyManagement>` section

File: service/pom.xml [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+            <groupId>org.springframework.security</groupId>
+            <artifactId>spring-security-crypto</artifactId>
+            <version>3.1.0.RELEASE</version>
+        </dependency>
+
+        <!--Repository Dependency-->
+        <dependency>
+            <groupId>com.example</groupId>
+            <artifactId>repository</artifactId>
+            <version>1.0-SNAPSHOT</version>
```
Comment (ensist) [inline service/pom.xml:31]: dependency's version is only set in parent POM, in `<dependencyManagement>` section

File: service/pom.xml [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+        </dependency>
+
+        <!--Web Model Dependency-->
+        <dependency>
+            <groupId>com.example</groupId>
+            <artifactId>webmodel</artifactId>
+            <version>1.0-SNAPSHOT</version>
```
Comment (ensist) [inline service/pom.xml:38]: dependency's version is only set in parent POM, in `<dependencyManagement>` section

File: service/pom.xml [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+        </dependency>
+
+        <!--Exceptions Dependency-->
+        <dependency>
+            <groupId>com.example</groupId>
+            <artifactId>exceptions</artifactId>
+            <version>1.0-SNAPSHOT</version>
```
Comment (ensist) [inline service/pom.xml:45]: dependency's version is only set in parent POM, in `<dependencyManagement>` section

File: webcontroller/pom.xml [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-webmvc</artifactId>
+            <version>5.1.0.RELEASE</version>
+        </dependency>
+
+        <!--Service Dependency-->
+        <dependency>
+            <groupId>com.example</groupId>
+            <artifactId>service</artifactId>
+            <version>1.0-SNAPSHOT</version>
```
Comment (ensist) [inline webcontroller/pom.xml:24]: dependency's version is only set in parent POM, in `<dependencyManagement>` section

File: webcontroller/pom.xml [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+        </dependency>
+
+        <!--Web Model Dependency-->
+        <dependency>
+            <groupId>com.example</groupId>
+            <artifactId>webmodel</artifactId>
+            <version>1.0-SNAPSHOT</version>
```
Comment (ensist) [inline webcontroller/pom.xml:31]: dependency's version is only set in parent POM, in `<dependencyManagement>` section

File: webmodel/src/main/java/com/example/webmodel/response/LoginResponseModel.java [hunk 17]
```diff
@@ -0,0 +1,11 @@
+package com.example.webmodel.response;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import java.math.BigInteger;
+
+@Getter @Setter
```
Comment (ensist) [inline webmodel/src/main/java/com/example/webmodel/response/LoginResponseModel.java:8]: change it to @Data

File: webmodel/src/main/java/com/example/webmodel/response/LoginResponseModel.java [hunk 18]
```diff
@@ -0,0 +1,11 @@
...
+public class LoginResponseModel {
+    private BigInteger _id;
```
Comment (ensist) [inline webmodel/src/main/java/com/example/webmodel/response/LoginResponseModel.java:10]: _id data type should be String

File: webmodel/src/main/java/com/example/webmodel/request/LoginRequestModel.java [hunk 19]
```diff
@@ -0,0 +1,9 @@
+package com.example.webmodel.request;
+
+import lombok.Getter;
+
+@Getter
+public class LoginRequestModel {
+    private String employeeUsername;
```
Comment (ensist) [inline webmodel/src/main/java/com/example/webmodel/request/LoginRequestModel.java:7]: remove prefix employee

File: webcontroller/src/main/java/com/example/webcontroller/LoginController.java [hunk 20]
```diff
@@ -0,0 +1,23 @@
...
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class LoginController {
+    @Autowired
+    private LoginService loginService;
+
+    @RequestMapping(value = "/login", method = RequestMethod.POST, produces = "application/json")
```
Comment (ensist) [inline webcontroller/src/main/java/com/example/webcontroller/LoginController.java:18]: move "/login" to new constant

Comment (ensist) [inline webcontroller/src/main/java/com/example/webcontroller/LoginController.java:18]: add consumes = "application/json"

File: webcontroller/src/main/java/com/example/webcontroller/LoginController.java [hunk 21]
```diff
@@ -0,0 +1,23 @@
...
+    public @ResponseBody
```
Comment (ensist) [inline webcontroller/src/main/java/com/example/webcontroller/LoginController.java:19]: no need @ResponseBody

File: webcontroller/src/main/java/com/example/webcontroller/LoginController.java [hunk 22]
```diff
@@ -0,0 +1,23 @@
...
+    LoginResponseModel callLoginService(@RequestBody LoginRequestModel loginRequestModel){
```
Comment (ensist) [inline webcontroller/src/main/java/com/example/webcontroller/LoginController.java:20]: should add BaseResponse like code, errorCode, errorMessage, value, etc

**Review by ensist — CHANGES_REQUESTED** [review]

File: .idea/.name (hunk 1 above)
Comment (jonathan016) [reply .idea/.name:1]: Commit ID: c03e75f

File: exception/src/main/java/com/example/exception/UserNotFoundException.java (hunk 2 above)
Comment (jonathan016) [reply exception/src/main/java/com/example/exception/UserNotFoundException.java:1]: Commit ID: e881ee0

Comment (jonathan016) [reply exception/src/main/java/com/example/exception/UserNotFoundException.java:1]: Commit ID: 468eeea

File: model/src/main/java/com/example/model/EmployeeModel.java (hunk 3 above)
Comment (jonathan016) [reply model/src/main/java/com/example/model/EmployeeModel.java:14]: Commit ID: a0d4316

File: model/src/main/java/com/example/model/EmployeeModel.java (hunk 4 above)
Comment (jonathan016) [reply model/src/main/java/com/example/model/EmployeeModel.java:13]: Commit ID: 103abaf

File: pom.xml (hunk 6 above)
Comment (jonathan016) [reply pom.xml:11]: Commit ID: 468eeea or e881ee0

File: webmodel/src/main/java/com/example/webmodel/request/LoginRequestModel.java (hunk 19 above)
Comment (jonathan016) [reply webmodel/src/main/java/com/example/webmodel/request/LoginRequestModel.java:7]: Commit ID: ada91dc

File: webcontroller/src/main/java/com/example/webcontroller/LoginController.java (hunk 20 above)
Comment (jonathan016) [reply webcontroller/src/main/java/com/example/webcontroller/LoginController.java:18]: Commit ID: ada91dc

Comment (jonathan016) [reply webcontroller/src/main/java/com/example/webcontroller/LoginController.java:18]: Commit ID: ada91dc

File: webcontroller/src/main/java/com/example/webcontroller/LoginController.java (hunk 21 above)
Comment (jonathan016) [reply webcontroller/src/main/java/com/example/webcontroller/LoginController.java:19]: Commit ID: cccaa6b

File: webmodel/src/main/java/com/example/webmodel/response/LoginResponseModel.java (hunk 18 above)
Comment (jonathan016) [reply webmodel/src/main/java/com/example/webmodel/response/LoginResponseModel.java:10]: Commit ID: 4d8251e

File: webmodel/src/main/java/com/example/webmodel/response/LoginResponseModel.java (hunk 17 above)
Comment (jonathan016) [reply webmodel/src/main/java/com/example/webmodel/response/LoginResponseModel.java:8]: Commit ID: 8cf4cf5

File: webcontroller/pom.xml (hunk 16 above)
Comment (jonathan016) [reply webcontroller/pom.xml:31]: Commit ID: 3092f2c

File: webcontroller/pom.xml (hunk 15 above)
Comment (jonathan016) [reply webcontroller/pom.xml:24]: Commit ID: 3092f2c

File: service/pom.xml (hunk 14 above)
Comment (jonathan016) [reply service/pom.xml:45]: Commit ID: 3092f2c

File: service/pom.xml (hunk 13 above)
Comment (jonathan016) [reply service/pom.xml:38]: Commit ID: 3092f2c

File: service/pom.xml (hunk 12 above)
Comment (jonathan016) [reply service/pom.xml:31]: Commit ID: 3092f2c

File: repository/pom.xml (hunk 11 above)
Comment (jonathan016) [reply repository/pom.xml:24]: Commit ID: 3092f2c

File: repository/pom.xml (hunk 10 above)
Comment (jonathan016) [reply repository/pom.xml:17]: Commit ID: 3092f2c

File: repository/pom.xml (hunk 9 above)
Comment (jonathan016) [reply repository/pom.xml:9]: Commit ID: 3092f2c

File: pom.xml (hunk 8 above)
Comment (jonathan016) [reply pom.xml:16]: Commit ID: 3092f2c

File: pom.xml (hunk 7 above)
Comment (jonathan016) [reply pom.xml:15]: Commit ID: 3092f2c

File: webcontroller/src/main/java/com/example/webcontroller/LoginController.java (hunk 22 above)
Comment (jonathan016) [reply webcontroller/src/main/java/com/example/webcontroller/LoginController.java:20]: Commit ID: f5c409a

File: model/src/main/java/com/example/model/EmployeeModel.java (hunk 5 above)
Comment (jonathan016) [reply model/src/main/java/com/example/model/EmployeeModel.java:20]: Commit ID: f5c409a
